actors/crypto/randomness.go

```diff
@@ -15,42 +16,55 @@ import (
 type DomainSeparationTag int
 
 const (
-	DomainSeparationTag_TicketDrawing DomainSeparationTag = 1 + iota
-	DomainSeparationTag_TicketProduction
+	DomainSeparationTag_TicketProduction DomainSeparationTag = 1 + iota
 	DomainSeparationTag_ElectionPoStChallengeSeed
 	DomainSeparationTag_SurprisePoStChallengeSeed
 	DomainSeparationTag_SurprisePoStSelectMiners
 	DomainSeparationTag_SurprisePoStSampleSectors
 )
 
 // Derive a random byte string from a domain separation tag and the appropriate values
-func DeriveRandWithMinerAddr(tag DomainSeparationTag, tix abi.RandomnessSeed, minerAddr addr.Address) abi.Randomness {
+func DeriveRandWithMinerAddr(tag DomainSeparationTag, randSeed []byte, minerAddr addr.Address) []byte {
 	var addrBuf bytes.Buffer
 	err := minerAddr.MarshalCBOR(&addrBuf)
 	autil.AssertNoError(err)
 
-	return _deriveRandInternal(tag, tix, -1, addrBuf.Bytes())
+	return _deriveRandInternal(tag, randSeed, addrBuf.Bytes())
 }
 
-func DeriveRandWithEpoch(tag DomainSeparationTag, tix abi.RandomnessSeed, epoch int64) abi.Randomness {
-	return _deriveRandInternal(tag, tix, -1, BigEndianBytesFromInt(epoch))
+type AddressEpochEntropy struct {
+	MinerAddress addr.Address // Must be an ID-addr
```

Comment:
We could extract the ID from the address prior to here, preventing the possibility of error. Other "must be an ID-addr" comments in the code are documenting preconditions for external callers.

Reply:
Agree, but I'd rather do that outside the randomness lib (ie with the external caller) rather than pass in the `initState` to the lib to do it here. That seems like a separate concern.